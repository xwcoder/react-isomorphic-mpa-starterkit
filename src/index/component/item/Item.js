import React from 'react'
import formatTime from '../../../lib/formatTime'

export default function Item ({ data }) {
  const {
    url,
    pubTime,
    title
  } = data

  return (
    <li className="list-item">
      <a href={url}>{title}</a>
      <span className="tag-time">{formatTime(pubTime, 'yyyy-MM-dd hh:mm:ss')}</span>
    </li>
  )
}
