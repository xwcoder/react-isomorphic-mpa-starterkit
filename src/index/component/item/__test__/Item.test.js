import React from 'react'
import { shallow } from 'enzyme'
import Item from '../Item'

describe('index/component/Item', () => {

  it('item with data', () => {
    const data = {
      docId: 1,
      url: 'http://localhost:8000/page/1',
      title: 'item test title',
      pubTime: 1522816928000
    }

    const wrapper = shallow(<Item data={data} />)

    expect(wrapper.find('a')).toHaveLength(1)
    expect(wrapper.find('.tag-time')).toHaveLength(1)
    expect(wrapper.find('a').prop('href')).toEqual(data.url)
    expect(wrapper.find('a').text()).toEqual(data.title)
    expect(wrapper).toMatchSnapshot()
  })
})
