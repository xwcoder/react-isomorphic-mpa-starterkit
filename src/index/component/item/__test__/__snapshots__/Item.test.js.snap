// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index/component/Item item with data 1`] = `
<li
  className="list-item"
>
  <a
    href="http://localhost:8000/page/1"
  >
    item test title
  </a>
  <span
    className="tag-time"
  >
    2018-04-04 12:42:08
  </span>
</li>
`;
